docs/src/DevelopersGuide.md

```diff
@@ -0,0 +1,142 @@
+    or one of the ``ci / ci 1.8.1 `` checks). Adjust the estimated minimum
+    runtime appropriately and this should fix your error.
+
+### GPU Tests
+
+GPU tests ensure that all code is able to run on GPU.
+    They can be found in the test folder under a single file named `gpu_tests.jl`.
+    There are two things to add: a kernel for your function and the actual test.
+
+Kernels are added at the top half of the file using `@kernel`.
+    The naming convention follows `test_<function name>_kernel!`. Within the kernel,
+    use `@inbounds` to place any outputs into an output array.
+
+The test itself should be added in the `test_gpu(FT)` function.
+    Look for the testset with the same name that the file your function is located in has.
+    You can copy and paste one of the other tests in the file to help guide you.
+    The test starts with defining `data_length` and ends after an `@test` macro.
+    `data_length` corresponds to the number of outputs your function has. The lines
+    that follow (up to the definition of `ndrange` can be left untouched). Add an
+    array for each input required by your function. For example, if you want to test at
+    temperature of 230K, you can add `T = ArrayType([FT(230)])`. Replace the kernel name
+    with your kernel on the `kernel!` line. On the `event` line, change the arguments of
+    `kernel!()` to match your kernel, keeping the last argument (`ndrange = ndrange`) untouched.
+    Add a comment of what you are testing and use `@test` to create your test. This is
+    typically making sure that your function runs on GPU so it can be something as simple as
+    checking that a certain input returns a known value.
+
+If you are receiving precision errors when running the `gpu_tests.jl`
+    file locally, it is likely because the tests are being ran twice: once with Float64
+    and again with Float32. This can be fixed by increasing the tolerance of your tests
+    or creating if-else statements.
+
+### Unit Tests
+
+Unit tests ensure that parameterizations make physical sense.
+    They can be found in the test folder under files named after corresponding source files.
+    If you created a new source file to add your function in, create a new test file that
+    would include unit tests for all functions in that new source file. If creating a new file
+    for unit tests, make sure you import `Test` and any other necessary libraries.
+
+Inside unit test functions, add parameters you would like to test
+    at and create separate testsets for each new function you've made. Add tests
+    using `@test` followed by a bool statement (i.e. 3 > 2, 19 = 19, 19.10001 ≈ 19.0).
+    Some possibile tests include checking if something is smaller/greater at warmer/cooler
+    temperatures, if assertion errors are returned when a function is used outside its
+    valid range of parameters, or if a function is zero at certain input values.
```

Comment:
```suggestion
    valid range of parameters, or if a function is zero at certain input values.
    Writing good tests id difficult, but at the very least we strive that all functions are exercised in some way in tests. So that at the very least we can catch changes in the API.
```